### PR TITLE
supported examples added

### PR DESCRIPTION
**Pull request by belous-dp**

### CODE REVIEW

File: integration-tests/c-example/lib/arrays.c [hunk 1]
```diff
@@ -1,27 +1,10 @@
-#include "multi_arrays.h"
```
Comment (ladisgin) [inline integration-tests/c-example/lib/arrays.c:1]: split to 2 files

File: integration-tests/c-example/lib/static.c [hunk 2]
```diff
@@ -1,49 +0,0 @@
-static int x = 0;
```
Comment (ladisgin) [inline integration-tests/c-example/lib/static.c:1]: don't delete this file just create issue with regression label

File: integration-tests/c-example/lib/loops.c [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    unsigned int i = 0;
+    do {
+        i = i + 1;
+        if (n % i == 37U)
+            return 1U;
+        else if (i == 50U)
+            return 2U;
+    } while (i < n);
+    return 0U;
+}
```
Comment (Lana243) [inline integration-tests/c-example/lib/loops.c:74]: Same as `while_loop`. I think that 2 of 3 functions can be removed.

File: integration-tests/c-example/lib/loops.c [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        i = i + 1;
+        if (n % i == 37)
+            return 1;
+        else if (i == 50)
+            return 2;
+    } while (i < n);
+    return 0;
+}
+
+unsigned int do_while_loop_unsigned_correct(unsigned int n) {
```
Comment (Lana243) [inline integration-tests/c-example/lib/loops.c:64]: if you decide to leave this function, add it to header, please.

File: integration-tests/c-example/lib/loops.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+int do_while_loop(int n);
+
+int do_while_loop_unsigned(unsigned int n);
+
+int continue_break(int n);
+
+int goto_keyword(unsigned int a);
+
+
```
Comment (Lana243) [inline integration-tests/c-example/lib/loops.h:16]: remove empty line

**Review by Lana243 — CHANGES_REQUESTED** [review]

File: integration-tests/c-example/lib/algorithm.c [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    }
+    return abs(x);
+}
+
+int sign(char const *str) {
+    int x = atoi(str);
+    if (x == 5) {
+        return 7;
+    }
+//    fprintf(stderr, "parsed int = %d\n", x);
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/algorithm.c:18]: done

File: integration-tests/c-example/lib/generic_selection.c [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        case 2:
+            return typeid(24);
+        case 3:
+            return typeid(42.f);
+        default:
+            return typeid("char const *");
+    }
+}
+
+
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/generic_selection.c:23]: done

File: integration-tests/c-example/lib/algorithm.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/algorithm.c:18]: done

File: integration-tests/c-example/lib/structures/simple_unions.h [hunk 8]
```diff
@@ -65,6 +65,13 @@ union UnionWithPointer {
     long long b;
 };
 
+struct WithAnonymous {
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/structures/simple_unions.h:68]: done

File: integration-tests/c-example/lib/structures/structs/complex_structs.h [hunk 9]
```diff
@@ -16,10 +16,17 @@ struct Three {
     struct Two in;
 };
 
-int struct_has_alphabet(struct One st);
+struct WithAnonymous {
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/structures/structs/complex_structs.h:19]: done

File: integration-tests/c-example/lib/structures/simple_unions.h [hunk 10]
```diff
@@ -73,6 +80,8 @@ int calculate_something_union(union Heterogeneous str);
 
 int get_coordinate(union Vector2D vect, int i);
 
+int count_equal_members(struct WithAnonymous st);
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/structures/simple_unions.h:83]: done

File: integration-tests/c-example/lib/structures/structs/complex_structs.h [hunk 11]
```diff
@@ -16,10 +16,17 @@ struct Three {
...
+    struct { int x, y; };
+    int m;
+};
 
+int struct_has_alphabet(struct One st);
 
 char arrays_in_inner_structs(struct Three st);
 
+int count_equal_members(struct WithAnonymous st);
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/structures/structs/complex_structs.h:28]: done

File: integration-tests/c-example/lib/multi_arrays.h [hunk 12]
```diff
@@ -1,7 +1,43 @@
...
+struct IntArray {
+    int ints[2][5];
+};
+
+struct PointStruct {
+    int x;
+    int y;
+};
+
+int get_00_elem(int a[1][2]);
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/multi_arrays.h:19]: done

File: integration-tests/c-example/lib/types/types.c [hunk 13]
```diff
@@ -54,11 +67,11 @@ bool is_positive(int arg) {
 }
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/types/types.c:68]: 😱 😱 😱 
done

File: integration-tests/c-example/lib/types/types.h [hunk 14]
```diff
@@ -1,64 +1,75 @@
...
 };
 
-struct SupportedStruct4 {
-    char* c;
+struct PtrCharStruct {
+    char *c;
 };
 
-struct SupportedStruct5 {
+struct AbcStruct {
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/types/types.h:28]: done

File: integration-tests/c-example/lib/types/types.c [hunk 15]
```diff
@@ -54,11 +67,11 @@ bool is_positive(int arg) {
...
 
-int supported_parameter_1(struct SupportedStruct1 st) {
+int supported_parameter_1(struct SimpleStruct st) {
     return 1;
 }
 
-int supported_parameter_2(struct SupportedStruct2 st) {
+int supported_parameter_2(struct LinkedStruct st) {
     return 2;
 }
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/types/types.c:77]: done

File: integration-tests/c-example/lib/loops.c [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+    unsigned int i = 0;
+    while (i < n) {
+        i = i + 1;
+        if (n % i == 37U)
+            return 1;
+        else if (i == 50U)
+            return 2U;
+    }
+    return 0U;
+}
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/loops.c:38]: see [issue 337](https://github.com/UnitTestBot/UTBotCpp/issues/337)

File: integration-tests/c-example/lib/loops.c [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+        i = i + 1;
+        if (n % i == 37)
+            return 1;
+        else if (i == 50)
+            return 2;
+    }
+    return 0;
+}
+
+unsigned int while_loop_unsigned_correct(unsigned int n) {
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/loops.c:28]: done

File: integration-tests/c-example/lib/loops.c (hunk 3 above)
Comment (belous-dp) [reply integration-tests/c-example/lib/loops.c:74]: see [previous response](https://github.com/UnitTestBot/UTBotCpp/pull/339#discussion_r932259672)

File: integration-tests/c-example/lib/loops.c (hunk 4 above)
Comment (belous-dp) [reply integration-tests/c-example/lib/loops.c:64]: done

File: integration-tests/c-example/lib/loops.h (hunk 5 above)
Comment (belous-dp) [reply integration-tests/c-example/lib/loops.h:16]: done

File: integration-tests/c-example/lib/loops.c (hunk 16 above)
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/loops.c:38]: fixed

File: integration-tests/c-example/lib/loops.c (hunk 3 above)
Comment (belous-dp) [reply integration-tests/c-example/lib/loops.c:74]: fixed

**Review by Lana243 — CHANGES_REQUESTED** [review]
Please add a single empty line at the end of each file. Also see comments in [unsupported examples PR](https://github.com/UnitTestBot/UTBotCpp-unsupported/pull/1) and move some examples here

**Comment by belous-dp** [comment]
> Please add a single empty line at the end of each file. Also see comments in [unsupported examples PR](https://github.com/UnitTestBot/UTBotCpp-unsupported/pull/1) and move some examples here

There's NL at the end of each file, github just doesn't show them. `pseudo_random` examples moved.

**Review by Lana243 — CHANGES_REQUESTED** [review]
Add example with static variable from unsupported repo: https://github.com/UnitTestBot/UTBotCpp-unsupported/pull/1#pullrequestreview-1063029557

File: integration-tests/c-example/lib/loops.c [hunk 18]
```diff
@@ -0,0 +1,56 @@
+#include "loops.h"
+#include <stdbool.h>
+
+unsigned int while_loop_unsigned_correct(unsigned int n) {
```
[the remark this answers is not in the document]
Comment (belous-dp) [reply integration-tests/c-example/lib/loops.c:4]: fixed

**Comment by belous-dp** [comment]
> Add example with static variable from unsupported repo: [UnitTestBot/UTBotCpp-unsupported#1 (review)](https://github.com/UnitTestBot/UTBotCpp-unsupported/pull/1#pullrequestreview-1063029557)

examples added

**Review by Lana243 — APPROVED** [review]
LGTM

**Review by ladisgin — APPROVED** [review]

**Pull request merged**